recipes/caterva/bld.bat

```diff
@@ -0,0 +1,27 @@
+setlocal EnableDelayedExpansion
+
+mkdir build
+if errorlevel 1 exit 1
+cd build
+if errorlevel 1 exit 1
+
+cmake -G "NMake Makefiles" ^
+      -DCMAKE_BUILD_TYPE:STRING="Release" ^
+      -DCMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" ^
+      -DCMAKE_PREFIX_PATH:PATH="%LIBRARY_PREFIX%" ^
+      -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON ^
+      -DSTATIC_LIB:BOOL=ON ^
```

Comment:
@FrancescAlted Here the static libs are still explicitly requested

Reply:
Oh yes, well spotted!  Now that I think, I am not sure that we should get rid of `.h` and `.a/.lib` files in the conda-forge package.  If we do, probably we need another `caterva-devel` conda-forge or similar so as to provide these for dependencies that need it. On the other hand, I suppose it is considered good practice to provide both packages.

Reply:
Headers (`.h`) should be included, just static libraries shouldn't be needed at all. For a package manager that manages the whole stack, it is much simpler to update versions when there is only dynamic linkage.

Reply:
Ok.  So I have disabled static libraries generation on both unix and win, and I am having issues with the Win platform not being able to generate `caterva.lib`:

```
Configuring for caterva version: 0.3.3
NMAKE : fatal error U1073: don't know how to make 'caterva.lib'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\nmake.exe"' : return code '0x2'
Stop.
```

If I remember correctly, `caterva.lib` is needed to link with `caterva.dll` on Win, but I lack the knowledge on how to generate `caterva.lib` without enabling static libraries, so I am going to unconditional enable static lib generation on win until we figure out how to properly do that via cmake.

Reply:
Can you add a step after the build where the *.a file is manually removed?

Reply:
I think the current approach is fine, although I'm a big fan of dont-ship-any-static-libs I understand that it building on Windows is a bit more complicated. 

@scopatz With MSVC, `.lib` is the static archive, there is no `.a` ;)

Reply:
Sounds good.  I am attempting to remove the *.lib (I think *.a is just meant for Unix).